gdx/src/com/badlogic/gdx/scenes/scene2d/ui/TextField.java

```diff
@@ -773,6 +770,28 @@ public void run () {
 		}
 	}
 
+	class CursorBlinkTask extends Task {
+		private boolean skipNextBlink = false;
+
+		@Override
+		public void run () {
+			if (skipNextBlink) {
+				skipNextBlink = false;
+				return;
+			}
+
+			cursorOn = !cursorOn;
+			Gdx.graphics.requestRendering();
+		}
+
+		/** Forces the task to skip next blink cycle update. Since {@link Timer} API doesn't provide a way to restart repeated task
+		 * immediately, this method is used to imitate blinking restart. */
```

Comment:
We should be able to cancel the task and schedule it again, then we don't need `skipNextBlink`.

Reply:
Nate, could you please implement `Timer#remove(Task)` or `Timer#restart(Task)` then it should be easy to change.

Reply:
I don't think we need that. Use:
```
Task task = ...
...
task.cancel();
Timer.instance().scheduleTask(task, delay, interval);
```
This works, though it will result in the task being in the timer task array 2+ times. The first one triggers the task, the rest are ignored, but it still isn't great. I have fixed that in a5fa372522445e23300ca8fe861dd384691b6932.